filament/src/components/LightManager.cpp

```diff
@@ -172,19 +172,24 @@ void FLightManager::create(const FLightManager::Builder& builder, utils::Entity
         lightType.lightCaster = builder->mCastLight;
 
         ShadowParams& shadowParams = manager[i].shadowParams;
-        shadowParams.options.mapSize = clamp(builder->mShadowOptions.mapSize, 0u, 2048u);
-        shadowParams.options.shadowCascades = clamp<uint8_t>(builder->mShadowOptions.shadowCascades, 1, CONFIG_MAX_SHADOW_CASCADES);
-        shadowParams.options.constantBias = clamp(builder->mShadowOptions.constantBias, 0.0f, 2.0f);
-        shadowParams.options.normalBias = clamp(builder->mShadowOptions.normalBias, 0.0f, 3.0f);
-        shadowParams.options.shadowFar = std::max(builder->mShadowOptions.shadowFar, 0.0f);
-        shadowParams.options.shadowNearHint = std::max(builder->mShadowOptions.shadowNearHint, 0.0f);
-        shadowParams.options.shadowFarHint = std::max(builder->mShadowOptions.shadowFarHint, 0.0f);
-        shadowParams.options.stable = builder->mShadowOptions.stable;
-        shadowParams.options.polygonOffsetConstant = builder->mShadowOptions.polygonOffsetConstant;
-        shadowParams.options.polygonOffsetSlope = builder->mShadowOptions.polygonOffsetSlope;
-        shadowParams.options.screenSpaceContactShadows = builder->mShadowOptions.screenSpaceContactShadows;
-        shadowParams.options.stepCount = builder->mShadowOptions.stepCount;
-        shadowParams.options.maxShadowDistance = builder->mShadowOptions.maxShadowDistance;
+        shadowParams.options = builder->mShadowOptions;
+
+        // validate all shadow options
+        shadowParams.options.mapSize = clamp(shadowParams.options.mapSize, 0u, 2048u);
```

Comment:
We could allow up to 4096 right?

Reply:
Yeah, but looking at the list of GPUs that support only [2048](https://opengles.gpuinfo.org/listreports.php?capability=GL_MAX_TEXTURE_SIZE&esversion=2&value=2048) and not 4096, I'm not sure it's worth having that limitation. Devices that support 2048 represent [3.7% of the total population](https://opengles.gpuinfo.org/displaycapability.php?name=GL_MAX_TEXTURE_SIZE&esversion=2). Everything else supports 4096 or more.